src/main/java/taxi/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,26 @@
+package taxi.service;
+
+import java.util.Optional;
+import taxi.dao.DriverDao;
+import taxi.exception.AuthenticationException;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverDao driverDao;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverDao.findByLogin(login);
+        if (driver.isEmpty()) {
+            throw new AuthenticationException("Login or password was incorrect");
```

Comment:
```suggestion
            throw new AuthenticationException("Login is incorrect");
```
Here you definitely know it, because the user was not found from the DB by the provided login.